src/main/java/bio/terra/service/auth/iam/IamService.java

```diff
@@ -148,6 +150,36 @@ public Map<UUID, Set<IamRole>> listAuthorizedResources(
     return callProvider(() -> iamProvider.listAuthorizedResources(userReq, iamResourceType));
   }
 
+  /**
+   * Note: calling this method will trigger a call to SAM API. No cache backs it. If calling,
+   * consider whether new usage patterns necessitate refactoring to add a cache.
+   *
+   * @param userReq authenticated user
+   * @param iamResourceType resource type; e.g. dataset
+   * @param resourceId UUID identifying a resource
+   * @param actions required
+   * @throws IamForbiddenException if the user is missing any `actions` on the resource
+   */
+  public void verifyAuthorizations(
+      AuthenticatedUserRequest userReq,
+      IamResourceType iamResourceType,
+      String resourceId,
+      List<IamAction> actions)
```

Comment:
I agree with you.  The one place where we cared about action order involved unit testing, but this was a good prompt to change the exception thrown by this method.  Instead of stringifying the missing action list in the message, we now share it as the `errorDetail` list.